scalikejdbc-streams/src/main/scala/scalikejdbc/streams/CloseableIterator.scala

```diff
@@ -0,0 +1,28 @@
+package scalikejdbc.streams
+
+import java.io.Closeable
+
+trait CloseableIterator[+A] extends Iterator[A] with Closeable {
+  self =>
+
+  override def close(): Unit
+
+  override def map[B](f: A => B): CloseableIterator[B] = new CloseableIterator[B] {
+    def hasNext: Boolean = self.hasNext
+    def next(): B = f(self.next())
+    def close(): Unit = self.close()
+  }
+}
+
+object CloseableIterator {
+
+  def empty[A]: CloseableIterator[A] = {
+    new CloseableIterator[A] {
+      override def close(): Unit = ()
+
+      override def next(): A = throw new IllegalStateException("an empty CloseableIterator.")
```

Comment:
https://github.com/scala/scala/blob/v2.12.1/src/library/scala/collection/Iterator.scala#L38

The same behavior is preferred. 

```scala
throw new NoSuchElementException("next on empty iterator")
```